### PR TITLE
Theme showcase: clean up themes/showcase-i4/search-and-filter flag

### PR DESCRIPTION
**Pull request by fushar**
#### Proposed Changes

This PR cleans up the `themes/showcase-i4/search-and-filter` flag as it is always true in every environment now.

#### Testing Instructions

Test that the following scenarios are still working correctly:

- Theme showcase for logged-out users
- Theme showcase for simple site
- Theme showcase for Jetpack site
- Theme showcase for Atomic site

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/my-sites/themes/logged-out.jsx [hunk 1]
```diff
@@ -8,9 +8,7 @@ const ConnectedThemeShowcase = connectOptions( ThemeShowcase );
 
 export default ( props ) => (
 	<Main fullWidthLayout className="themes">
-		{ isEnabled( 'themes/showcase-i4/search-and-filter' ) && (
-			<BodySectionCssClass bodyClass={ [ 'is-section-themes-i4' ] } />
-		) }
+		<BodySectionCssClass section="themes" />
```
Comment (taipeicoder) [inline client/my-sites/themes/logged-out.jsx]: IIRC the page already has the class name `is-section-themes` added, so we don't need to explicitly add it here. Could you confirm if this is the case?

File: client/my-sites/themes/single-site-jetpack.jsx [hunk 2]
```diff
@@ -75,19 +73,13 @@ const ConnectedSingleSiteJetpack = connectOptions( ( props ) => {
 
 	return (
 		<Main fullWidthLayout className="themes">
-			{ isNewSearchAndFilter && <BodySectionCssClass bodyClass={ [ 'is-section-themes-i4' ] } /> }
+			<BodySectionCssClass section="themes" />
```
Comment (taipeicoder) [inline client/my-sites/themes/single-site-jetpack.jsx]: Same here, could you check if this is necessary?

File: client/my-sites/themes/single-site-wpcom.jsx [hunk 3]
```diff
@@ -81,19 +79,13 @@ const ConnectedSingleSiteWpcom = connectOptions( ( props ) => {
 
 	return (
 		<Main fullWidthLayout className="themes">
-			{ isNewSearchAndFilter && <BodySectionCssClass bodyClass={ [ 'is-section-themes-i4' ] } /> }
+			<BodySectionCssClass section="themes" />
```
Comment (taipeicoder) [inline client/my-sites/themes/single-site-wpcom.jsx]: Same here 🙂

**Review by taipeicoder — APPROVED** [review]
Tested:
- Logged in with simple site.
- Logged in with atomic site.
- Logged out.

Works great, thanks for this 🤩
The only feedback I had was the use of `<BodySectionCssClass/>` and whether we need it or not. Other than that, all good on my end 👍

File: client/my-sites/themes/logged-out.jsx (hunk 1 above)
Comment (fushar) [reply client/my-sites/themes/logged-out.jsx]: Thanks, you are right! Removed :)

**Pull request merged**